src/xma/include/lib/xmahw_lib.h

```diff
@@ -95,6 +98,49 @@ typedef struct XmaHwExecBO
   }
 } XmaHwExecBO;
 
+typedef struct XmaBufferPool
+{
+    std::list<XmaBufferObj>   buffers_busy;
+    std::queue<XmaBufferObj>  buffers_free;
+    std::atomic<bool> pool_locked;
```

Comment:
Why atomic here.  Shouldn't the pool be using a mutex to ensure atomic operations?

Reply:
I think in this case mutex can be avoided. atomic is only for internal synchronization. Will add mutex if it can not be avoided.

Reply:
An atomic that indicates that a structure is locked cannot be more efficient that using a mutex to ensure mutual exclusion.  I am not understanding.